dts/bindings/input/espressif,touch-sensor.yaml

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Zephyr input touch sensor parent node
+
+  This defines a group of touch sensors that can generate input events. Each touch
+  sensor is defined in a child node of the touch-sensor node and defines a specific key
+  code.
+
+  For example:
+
+  #include <zephyr/dt-bindings/input/input-event-codes.h>
+  #include <zephyr/dt-bindings/input/esp32-touch-sensor-input.h>
+
+  / {
+         touch_sensors {
```

Comment:
question: isn't this a MMIO device? if so, it should have an address, and be placed under `soc/` DT node (simple-bus). likely skipped here because of HAL...

Reply:
touch sensor is not a MMIO in the same sense as a GPIO. Besides that our hal implements a behavior near to an external interrupt input by software, so we used hal functions to generate events and publish the on input subsystem.

Reply:
are you saying such IP doesn't have an address? where does the HAL write to?

Reply:
you are right, at some point hal functions have to touch some memory mapped registers.
I'll provide an entry under /soc DT section for each soc with touch_sensor peripheral.